src/transforms/scaling.jl

```diff
@@ -0,0 +1,58 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Scaling(sx, sy, sz)
+
+Transform geometry or mesh by scaling each coordinate of the vertices by 
+the corresponding factor. 
+
+## Arguments
+
+- `sx`: scaling factor for the first coordinate
+- `sy`: scaling factor for the second coordinate
+- `sz`: scaling factor for the third coordinate
+
+## Examples
+
+```julia
+Scaling(1, 2, 3)
+```
+"""
+struct Scaling{T<:Real} <: GeometricTransform
+  sx::T
+  sy::T
+  sz::T
+end
+
+function Scaling(sx::T1, sy::T2, sz::T3) where {T1<:Real,T2<:Real,T3<:Real}
+  if sx <= 0 || sy <= 0 || sz <= 0
+    error("The scaling factors must be nonnegative.")
+  end
+  T = promote(sx, sy, sz)
```

Comment:
Probably does not compile? Can you please make sure that the code in your PRs is compiling before submitting it for review?

Reply:
It compiles, I ran the unit tests.

Reply:
You are using promote incorrectly again. I don't know how you are getting the correct results because the promote function returns a tuple of numbers and you are using T as a type. Please, please, please... run these codes locally before submitting the PR. The time here is really scarce to review code that has not been even run by the person who submitted the PR. Keep in mind that this project is used by many people and we don't want serious bugs crashing engineering systems out there.

Reply:
Weird, you're right. I always run the unit tests before submitting. I don't understand why they didn't crash here!

Reply:
The unit tests will only crash if you add new tests covering the new functionality. Otherwise you will just be testing old code that is already known. Every PR usually comes with a new set of tests. This one here is missing new tests.

Reply:
Are you joking? ;-) There's a `@testset "Scaling"` in this PR, and I ran it.

Reply:
Ah that is true, then I don't understand how tests could have passed. The code is clearly incorrect treating T as a type when it is a tuple of numbers.

Reply:
I triggered the tests on GitHub again to see what happens.

Reply:
Is it possible that `Scaling(T(1), T(2), T(3))` has applied the `struct` and not my function? I don't see any other explanation. Not sure I'm clear.